### PR TITLE
BE: Signup / Login API Routes

### PR DESCRIPTION
**Pull request by perryraskin**
Implements Registration and Authorization API along with User model for mongoose

### CODE REVIEW

**Comment by perryraskin** [comment]
Fixed comment in `/api/auth.js` for the `GET` request. 

```
// @route   POST api/auth/user
```

It says `POST` instead of `GET`. Will commit that change after, unless there's an easy way to add it to this PR?

File: server/routes/api/auth.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// @route   POST api/auth
+// @desc    Auth user
+// @access  Public
+router.post("/", function(req, res, next) {
+  const { email, password } = req.body;
+  try {
+    const normalizedEmail = normalizeEmail(email);
+    if (!isEmail(normalizedEmail)) throw new Error('The email you entered is invalid');
+    if (!password || !email) throw new Error('Missing field(s)');
+    const date = new Date();
```
Comment (bonnieli) [inline server/routes/api/auth.js]: is this used?

**Review by bonnieli — COMMENTED** [review]
overall looks good

File: server/routes/api/auth.js (hunk 1 above)
Comment (perryraskin) [reply server/routes/api/auth.js]: Yeah, this is for input validation for a login page

Comment (perryraskin) [reply server/routes/api/auth.js]: Oh but I'll remove the `date`, which is not used

**Review by patelparth1993 — APPROVED** [review]
Looks fine.

**Pull request merged**